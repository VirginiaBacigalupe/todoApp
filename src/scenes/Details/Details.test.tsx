import { useRoute } from '@react-navigation/native'
import { shallow } from 'enzyme'
import React from 'react'
import { Text } from 'react-native'

import { TaskType } from '../..'
import { Details } from './index'

jest.mock('@react-navigation/native')
jest.mock('react-redux')

const mockUseRoute = useRoute as jest.Mock

const task: TaskType = {
  completed: false,
  description: 'desc tarea 1',
  id: '1',
  title: 'tarea1',
}

describe('Details screen', () => {
  it('render ok', () => {
    mockUseRoute.mockReturnValue({ params: { task } })
    const wrapper = shallow(<Details></Details>)
    expect(wrapper).toMatchSnapshot()

    expect(wrapper.find(Text)).toHaveLength(3)

    const taskStatusText = wrapper.find('[testID="taskStatus"]').length
    expect(taskStatusText).toEqual(1)
    expect(taskStatusText.valueOf)

    const taskTitleText = wrapper.find('[testID="taskTitle"]').length
    expect(taskTitleText).toEqual(1)

    const taskDescriptionText = wrapper.find(
      '[testID="taskDescription"]',
    ).length
    expect(taskDescriptionText).toEqual(1)
  })
})
