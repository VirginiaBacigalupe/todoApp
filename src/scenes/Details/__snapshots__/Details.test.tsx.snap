// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details screen render ok 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(255, 255, 255)",
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(255, 25, 123)",
        "margin": 10,
        "marginBottom": 5,
      }
    }
    testID="taskStatus"
  >
    Not done
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(0,0,0)",
        "fontSize": 36,
        "margin": 10,
        "marginTop": 5,
      }
    }
    testID="taskTitle"
  >
    tarea1
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(149, 149, 149)",
        "fontSize": 14,
        "margin": 10,
      }
    }
    testID="taskDescription"
  >
    desc tarea 1
  </Text>
  <PinkButton
    onPress={[Function]}
    title="MARK AS DONE"
  />
</View>
`;
